### ISSUE TITLE
The result of source only ResNet-50

### ISSUE THREAD

**Issue by HHHedo** [jihanyang/AFN#11]
Hi jihan,
It's a great work, and I am now reproducing the results. 
I am a little bit confused about the result of the source only ResNet-50. I believe this result can be achieved by simply setting the weights of L2_norm loss and ent loss to be zero. However, it turned out that the results are much better. For example, in Office-31 A to W, the accuracy is about 0.79 (0.68 reported). Though, I saw that the same results are also reported in other papers. 
Looking forward to your reply.

**Comment by jihanyang** [comment]
hello, thanks for your interesting to our work. 
To reproduce the source only result of ResNet-50, 1. you need to remove our L_2-dropout in forward stage. 2. you need to remove the bottleneck layer in our classifier. This layer is widely used in previous methods (eg., CDAN, JAN etc.), while doesn't appear in original ResNet-50 architecture. 
Please try these manners first to address your problem. If you have any other questions, feel free to cantact us. Hope my suggestion is useful for you.

Jihan

**Comment by HHHedo** [comment]
Hi,
Does "the bottleneck layer in classifier" mean the bottleneck in residual blocks should be replaced by basicblocks?
In Pytorch, the implement is 
`resnet('resnet50', Basicblock, [3, 4, 6, 3], pretrained, progress,  **kwargs)`

**Comment by jihanyang** [comment]
Hello, you can check the difference between the  classifier of pytorch offical ResNet-50 and ours. 
https://github.com/jihanyang/AFN/blob/d96515b26223278f51f8e02477a42dbc5f13bef9/vanilla/Visda2017/SAFN/code/model/net.py#L68-L79

You can also check the bottleneck layer implementation of CDAN.
https://github.com/thuml/CDAN/blob/f7889063b76fca0b9a7147c88103d356531924bd/pytorch/network.py#L137

**Comment by HHHedo** [comment]
Hi,
I just modified the classifier as you advised, which is
` self.fc3 = nn.Linear(2048, class_num)`
And the result for Office-31 A to W is:
`Epoch 50: 0.740880503144654
Epoch 60: 0.7534591194968554
Epoch 70: 0.7509433962264151
Epoch 80: 0.7257861635220125
Epoch 90: 0.7584905660377359
Epoch 100: 0.7446540880503144
`
which is still better than reported.

**Comment by jihanyang** [comment]
Hello, you shouldn't use target images for source only result.

**Comment by HHHedo** [comment]
> Hello, you shouldn't use target images for source only result.

Of course, I didn't use the target images. 
where the main code is :
`opt_g.zero_grad()`
`s_bottleneck = netG(s_imgs)`
`s_logit = netF(s_bottleneck)`
`s_cls_loss = get_cls_loss(s_logit, s_labels)`
`loss = s_cls_loss`
` loss.backward()`
` opt_g.step()`
Maybe you can try this yourself, though this has nothing to do with AFN which can be reporduced.
Anyway, many thanks to your patient answers and help.

**Comment by jihanyang** [comment]
Hello, due to my recent projects, I am sorry that I cannot schedule time to check the result. I will try to reproduce this experiment once I finish my projects at hand.

**Comment by Ruijia-Xu** [comment]
Hello @HHHedo, thanks first for your interests in our work.

You are right by pointing out the fact that most of existing papers (including this work) share a same copy of the reported accuracies under the 'source-only' setting. 

What might make you confused is the exact object this 'source-only' model point to. In these papers, it means a model that uses the source domain to fine-tune the Imagenet-pretrained model and then directly deploys in the target domain. It was conducted and reported by the very first paper (I forget which paper...) and cited by other papers.  

However, as different DA methods develop, you might have to adjust the operators (e.g., bn, dropout...), configuration of the task-specific classifier, optimization strategy, ..., which differ significantly from the shape of the original 'source-only' model. As a result, the latter one cannot be directly obtained by just excluding some losses.

From my own view, it is also encouraged to report both results, one from the original paper and the other one by using your own practices. 

Best.

**Comment by HHHedo** [comment]
Many thanks to your help, and my doubts are solved.
It’s very kind of you!

Best
